src/r2mm/installing/profile_installers/GenericProfileInstaller.ts

```diff
@@ -286,6 +287,40 @@ export default class GenericProfileInstaller extends ProfileInstallerProvider {
         }
     }
 
+    private async installPackageZip(profile: Profile, rule: ManagedRule, installSources: string[], mod: ManifestV2) {
+        /*
+            This install method repackages the entire mod as-is and places it to the
+            destination route. Essentially the same as SUBDIR_NO_FLATTEN, but as a
+            zip instead of a directory. The zip name will be the mod ID.
+         */
+        const destDir = path.join(profile.getPathOfProfile(), rule.route);
+        await FileUtils.ensureDirectory(destDir);
+        const destination = path.join(destDir, `${mod.getName()}.ts.zip`);
+        const cacheDirectory = path.join(PathResolver.MOD_ROOT, 'cache');
+        const cachedLocationOfMod: string = path.join(cacheDirectory, mod.getName(), mod.getVersionNumber().toString());
+        const builder = ZipProvider.instance.zipBuilder();
+        await builder.addFolder("", cachedLocationOfMod);
+        await builder.createZip(destination);
```

Comment:
Not much of a concern right now but it may be nice to have this stored somewhere in the cache so we don't need to do it each time. One issue is storage space, but generally I think it would help to speed up reinstalls across profiles as zipping larger mods may take a while.

Reply:
I considered it but left it out for the sake of code simplicity. I figured we'll eventually rework this area and can account for it when that happens (be it just refactoring the current installer or migrating to TCLI).